File: Wisdom-Knowledge/serena/Serena.java
public class Serena {
    private final String name = "Serena";
    private int magicPower;
    private int mana;

    public Serena(int magicPower, int mana) {
        this.magicPower = magicPower;
        this.mana = mana;
    }

    public int orbOfBeyond() {
        return 200 + (int)(1.6 * magicPower);
    }

    public String codexOfWisdom(String target) {
        return "Hypnotized " + target + " for 3 seconds. It now fights for Serena.";
    }

    public int eternalCircle() {
        return 300 + (int)(1.8 * magicPower);
    }

    public String eternum(int targetHealthPercent, boolean receivedHeal) {
        if (targetHealthPercent <= 15) {
            return "Target executed!";
        } else if (receivedHeal) {
            return "Target receives burning damage for 3 seconds.";
        } else {
            return "Target survived the magical blast.";
        }
    }
}

File: Wisdom-Knowledge/serena/serena.py
class Serena:
    def __init__(self, magic_power, mana):
        self.name = "Serena"
        self.magic_power = magic_power
        self.mana = mana

    def orb_of_beyond(self):
        return 200 + int(1.6 * self.magic_power)

    def codex_of_wisdom(self, target):
        return f"Hypnotized {target} for 3 seconds. It now fights for Serena."

    def eternal_circle(self):
        return 300 + int(1.8 * self.magic_power)

    def eternum(self, target_health_percent, received_heal):
        if target_health_percent <= 15:
            return "Target executed!"
        elif received_heal:
            return "Target receives burning damage for 3 seconds."
        else:
            return "Target survived the magical blast."
